### PR TITLE
[lldb] Convert more tests to use `expression` directly (NFC)

### PR DESCRIPTION
**Pull request by kastiglione**
Convert more uses of `p` and `print`, following on the initial conversion done in [D141539](https://reviews.llvm.org/D141539).

This also cherry picks [D146230](https://reviews.llvm.org/D146230)

### CODE REVIEW

**Comment by kastiglione** [comment]
@swift-ci test

File: lldb/test/API/commands/expression/expr_inside_lambda/TestExprInsideLambdas.py [hunk 1]
```diff
@@ -54,7 +54,7 @@ def test_expr_inside_lambda(self):
         self.expect_expr("base_var", result_type="int", result_value="14")
         self.expect_expr("this->shadowed", result_type="int", result_value="-1")
         
-        # 'p this' should yield 'struct Foo*'
+        # 'expression this' should yield 'struct Foo*'
```
Comment (adrian-prantl) [inline lldb/test/API/commands/expression/expr_inside_lambda/TestExprInsideLambdas.py]: Did you make these changes upstream, too?

Comment (adrian-prantl) [reply lldb/test/API/commands/expression/expr_inside_lambda/TestExprInsideLambdas.py]: You may need to resolve fewer merge conflicts if you cherry-pick the change first

Comment (kastiglione) [reply lldb/test/API/commands/expression/expr_inside_lambda/TestExprInsideLambdas.py]: I backed this change out. Will do upstream.

**Comment by kastiglione** [comment]
@swift-ci test

**Comment by kastiglione** [comment]
@swift-ci test

**Pull request merged**